### PR TITLE
Smartling Downloader Modifications

### PR DESCRIPTION
**Pull request by bencgreenberg**
## Description

Compare original string and word count in the original file against the completed string and word count for each locale and only download fully published files that match.

### CODE REVIEW

**Comment by mheap** [comment]
> According to the latest correspondence with Smartling support, adding the includeOriginalStrings boolean param to the download request with a value of false will eliminate the partially completed translation returns. Testing this in this branch of Station to see if that is the case.

This doesn't sound right to me. Looking at their API docs

image

This does not stop us downloading partially translated files. It will strip non-translated content out, which could lead to misleading documentation

**Comment by bencgreenberg** [comment]
> > According to the latest correspondence with Smartling support, adding the includeOriginalStrings boolean param to the download request with a value of false will eliminate the partially completed translation returns. Testing this in this branch of Station to see if that is the case.
> 
> This doesn't sound right to me. Looking at their API docs
> 
> image
> 
> This does not stop us downloading partially translated files. It will strip non-translated content out, which could lead to misleading documentation

Indeed, according to their documentation that is what it does. However, their support team confirms that their docs are not accurate. It removes content that is partially translated, and it also supports markdown in contrast to their docs, once again.

**Comment by bencgreenberg** [comment]
@fabianrbz I've tested this today against the Smartling job for the documentation index page and it works. 🎉

File: lib/nexmo_developer/app/services/translator/smartling/api/download_file.rb [hunk 1]
```diff
@@ -18,7 +18,13 @@ def build_request
         def uri
           @uri ||= begin
             uri = URI("https://api.smartling.com/files-api/v2/projects/#{@project_id}/locales/#{@locale_id}/file")
-            uri.query = URI.encode_www_form({ 'fileUri' => @file_uri, 'retrievalType' => 'published' })
+            uri.query = URI.encode_www_form(
+              {
+                'fileUri' => @file_uri,
+                'retrievalType' => 'published',
+                'includeOriginalStrings' => false,
```
Comment (fabianrbz) [inline lib/nexmo_developer/app/services/translator/smartling/api/download_file.rb:25]: @bencgreenberg do we still need the `includeOriginalStrings` flag?

Comment (bencgreenberg) [reply lib/nexmo_developer/app/services/translator/smartling/api/download_file.rb:25]: No, we don't, great catch! Let me get rid of it, one sec.

**Review by fabianrbz — APPROVED** [review]

**Pull request merged**